src/test/kotlin/pt/up/fe/ni/website/backend/utils/listeners/DbCleanupListener.kt

```diff
@@ -0,0 +1,99 @@
+package pt.up.fe.ni.website.backend.utils.listeners
+
+import jakarta.persistence.EntityManager
+import jakarta.transaction.Transactional
+import org.hibernate.id.enhanced.PooledOptimizer
+import org.hibernate.id.enhanced.SequenceStyleGenerator
+import org.hibernate.metamodel.model.domain.internal.MappingMetamodelImpl
+import org.springframework.test.context.TestContext
+import org.springframework.test.context.TestExecutionListener
+import java.sql.Connection
+import java.sql.Statement
+import javax.sql.DataSource
+
+class DbCleanupListener : TestExecutionListener {
+    companion object {
+        private const val SCHEMA_NAME = "PUBLIC"
+        private const val DB_NAME = "H2"
+    }
+
+    override fun beforeTestMethod(testContext: TestContext) {
+        super.beforeTestMethod(testContext)
+        println("Cleaning up database...")
```

Comment:
Does this show up in the test log?

Reply:
It does

Reply:
Should I change anything or can we merge? This is an important PR @DoStini

Reply:
Sorry for the delay, it might be a bit cluttered but we can merge